packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -268,143 +278,59 @@ exports.sourceNodes = async (
-            } else {
-              console.error(`Failed to fetch ${url}`, error.message)
-              console.log(error)
-              throw error
-            }
-          }
-          dataArray.push(...d.body.data)
-          // Add support for includes. Includes allow entity data to be expanded
-          // based on relationships. The expanded data is exposed as `included`
-          // in the JSON API response.
-          // See https://www.drupal.org/docs/8/modules/jsonapi/includes
-          if (d.body.included) {
-            dataArray.push(...d.body.included)
-          }
-          if (d.body.links && d.body.links.next) {
-            await getNext(d.body.links.next)
+  const listResponse = await requestQueue.push([
+    urlJoin(baseUrl, 'gatsby/content-list'),
+    {
+      headers,
+    }
+  ])
+  const listResponseBody = JSON.parse(listResponse.body)
+    
+  const requestPromises = []
+  for (let entityTypeAndBundle in listResponseBody) {
+    if (disallowedLinkTypes.indexOf(entityTypeAndBundle) !== -1) continue
+    const isTranslatable = languageConfig.translatableEntities.indexOf(entityTypeAndBundle) !== -1
+    const shouldUseAuth = useAuthOn.indexOf(entityTypeAndBundle) !== -1
+    const [entityType, entityBundle] = entityTypeAndBundle.split('--')
+
+    for (let entityUuid of listResponseBody[entityTypeAndBundle]) {
+      requestPromises.push(
+        requestQueue.push([
+          urlJoin(baseUrl, apiBase, `/${entityType}/${entityBundle}/${entityUuid}`),
+          {
+            headers: shouldUseAuth ? headers : undefined,
           }
-        }
-
-        if (isTranslatable === false) {
-          await getNext(url)
-        } else {
-          for (let i = 0; i < languageConfig.enabledLanguages.length; i++) {
-            let currentLanguage = languageConfig.enabledLanguages[i]
-            const urlPath = url.href.split(`${apiBase}/`).pop()
-            const baseUrlWithoutTrailingSlash = baseUrl.replace(/\/$/, ``)
-            // The default language's JSON API is at the root.
-            if (
-              currentLanguage === getOptions().languageConfig.defaultLanguage ||
-              baseUrlWithoutTrailingSlash.slice(-currentLanguage.length) ==
-                currentLanguage
-            ) {
-              currentLanguage = ``
-            }
+        ]).then(response => JSON.parse(response.body).data).catch(() => {})
+      )
 
-            const joinedUrl = urlJoin(
-              baseUrlWithoutTrailingSlash,
-              currentLanguage,
-              apiBase,
-              urlPath
+      if (isTranslatable) {
+        for (let language of languageConfig.enabledLanguages) {
+          if (language !== languageConfig.defaultLanguage) {
+            requestPromises.push(
+              requestQueue.push([
+                urlJoin(baseUrl, language, apiBase, `/${entityType}/${entityBundle}/${entityUuid}`),
+                {
+                  headers: shouldUseAuth ? headers : undefined
```

Comment:
This dumps all the other headers if you don't want auth on the entity, need a better way to do this.